mlir/lib/Conversion/RaiseWasm/RaiseWasmMLIR.cpp

```diff
@@ -42,19 +44,17 @@ using namespace mlir::wasm;
 
 namespace {
 
-template <typename SourceOp, typename TargetIntOp, typename TargetFPOp>
+template <typename SourceOp, typename TargetIntOp, typename TargetFPOp, typename IntrinsicOp>
```

Comment:
I think the logic here is rather complicated.
What about having a separable template pattern for arithmetic ops that traps?